### DISCUSSION TITLE
CodeQL DB from apk

### DISCUSSION THREAD

**Discussion by boichique** [github/codeql#13791] [General]
Hello there!

Can please someone give me advice about my problem? I need to create CodeQL database for apk file. I understand that I need to decompile apk first (doing this with JADX, save as gradle project), but CodeQL still can't create database due to compile errors...

**Comment by jketema** [comment]
Hi @boichique,

Thanks for your question. Does the decompiled code compile?

**Reply by boichique** [reply]
No, the decompiled code does not compile... That's why I decided to ask for advice here :)

**Reply by jketema** [reply]
We assume that, if you take CodeQL out of the loop, then the code compiles. If it does not, there's not much we guarantee from our side.

**Reply by atorralba** [reply]
Also, note the [License Restrictions](https://github.com/github/codeql-cli-binaries/blob/main/LICENSE.md#license-restrictions) of CodeQL.

**Reply by boichique** [reply]
@jketema That's fair
